File: static/js/IntViolation.js

function reset_IntViolation(){
    $('#form_action').val('add');
    $('#IntViolation_id').val('');

    $('#IntViolation_kootaj').val('');
    $('#IntViolation_cert_no').val('');
    $('#IntViolation_s_date').val('');
    $('#IntViolation_m_date').val('');
    $('#IntViolation_int_cert_no').val('');
    $('#IntViolation_radif_marzi').val('');
    $('#IntViolation_s_entry_date').val('');
    $('#IntViolation_m_entry_date').val('');
    $('#IntViolation_karne_tir_no').val('');
    $('#IntViolation_ezhar_1').val('');
    $('#IntViolation_ezhar_2').val('');
    $('#IntViolation_ezhar_3').val('');
    $('#IntViolation_ezhar_4').val('');
    $('#IntViolation_person_type').val('');
    $('#IntViolation_full_name').val('');
    $('#IntViolation_card_no').val('');
    $('#IntViolation_code_no').val('');
    $('#IntViolation_dest_address').val('');
    $('#IntViolation_good_bame').val('');
    $('#IntViolation_net_weight').val('');
    $('#IntViolation_weight').val('');
    $('#IntViolation_input_tariff').val('');
    $('#IntViolation_good_count').val('');
    $('#IntViolation_price').val('');
    $('#IntViolation_truck_no').val('');
    $('#IntViolation_driver_name').val('');
    $('#IntViolation_chasis_no').val('');
    $('#IntViolation_trailer_no').val('');
    $('#IntViolation_tranian_truck').val('');
    $('#IntViolation_truck_address').val('');
    $('#IntViolation_truck_type').val('');
    $('#IntViolation_vi_type_1').val('');
    $('#IntViolation_vi_type_2').val('');
    $('#IntViolation_vi_type_3').val('');
    $('#IntViolation_vi_type_4').val('');
    $('#IntViolation_vi_type_5').val('');
    $('#IntViolation_vi_type_6').val('');
    $('#IntViolation_vi_type_7').val('');
    $('#IntViolation_detector').val('');
    $('#IntViolation_detector_other').val('');
    $('#IntViolation_fine_receipt_no').val('');
    $('#IntViolation_fine_price').val('');
    $('#IntViolation_final_result').val('');
    $('#IntViolation_details').val('');
    $('#IntViolation_m_rec_date').val('');
    $('#IntViolation_s_rec_date').val('');
    $('#IntViolation_locked').val('');

}

function edit_IntViolation(id){
    $('#form_action').val('info');
    $('#IntViolation_id').val(id);
    submit_IntViolation();
}

function remove_IntViolation(id){
    $('#form_action').val('delete');
    $('#IntViolation_id').val(id);

    if (confirm('are you sure')){
        submit_IntViolation();
    }
}

function submit_IntViolation(){
    var f_action = $('#form_action').val()
    if (f_action != 'delete' && f_action != 'info'){
        // validation
    }
    var form = $('#form_IntViolation');
    $.ajax({
        url: '',
        type: 'POST',
        data: form.serialize(),
        cache: false
    }).done(function(responce){
        var d = JSON.parse(responce);
        if (d.ret == 'error' || d.action == 'error'){
            alert('Can not complete action. ' + d.ret);
            return;
        }
        switch (d.action){
            case 'add':
                var row = parseInt($('#table_IntViolation tr:last td:first').html());
                row = (!row)? 1 : row + 1;
                var td ='<tr id="IntViolation_' + d.ret + '">' +
                        '<td>' + row + '</td>' +
                        '<td id="IntViolation_kootaj_' + d.ret + '">' + d.full_data.kootaj + '</td>' +
                        '<td id="IntViolation_cert_no_' + d.ret + '">' + d.full_data.cert_no + '</td>' +
                        '<td id="IntViolation_s_date_' + d.ret + '">' + d.full_data.s_date + '</td>' +
                        '<td id="IntViolation_m_date_' + d.ret + '">' + d.full_data.m_date + '</td>' +
                        '<td id="IntViolation_int_cert_no_' + d.ret + '">' + d.full_data.int_cert_no + '</td>' +
                        '<td id="IntViolation_radif_marzi_' + d.ret + '">' + d.full_data.radif_marzi + '</td>' +
                        '<td id="IntViolation_s_entry_date_' + d.ret + '">' + d.full_data.s_entry_date + '</td>' +
                        '<td id="IntViolation_m_entry_date_' + d.ret + '">' + d.full_data.m_entry_date + '</td>' +
                        '<td id="IntViolation_karne_tir_no_' + d.ret + '">' + d.full_data.karne_tir_no + '</td>' +
                        '<td id="IntViolation_ezhar_1_' + d.ret + '">' + d.full_data.ezhar_1 + '</td>' +
                        '<td id="IntViolation_ezhar_2_' + d.ret + '">' + d.full_data.ezhar_2 + '</td>' +
                        '<td id="IntViolation_ezhar_3_' + d.ret + '">' + d.full_data.ezhar_3 + '</td>' +
                        '<td id="IntViolation_ezhar_4_' + d.ret + '">' + d.full_data.ezhar_4 + '</td>' +
                        '<td id="IntViolation_person_type_' + d.ret + '">' + d.full_data.person_type + '</td>' +
                        '<td id="IntViolation_full_name_' + d.ret + '">' + d.full_data.full_name + '</td>' +
                        '<td id="IntViolation_card_no_' + d.ret + '">' + d.full_data.card_no + '</td>' +
                        '<td id="IntViolation_code_no_' + d.ret + '">' + d.full_data.code_no + '</td>' +
                        '<td id="IntViolation_dest_address_' + d.ret + '">' + d.full_data.dest_address + '</td>' +
                        '<td id="IntViolation_good_bame_' + d.ret + '">' + d.full_data.good_bame + '</td>' +
                        '<td id="IntViolation_net_weight_' + d.ret + '">' + d.full_data.net_weight + '</td>' +
                        '<td id="IntViolation_weight_' + d.ret + '">' + d.full_data.weight + '</td>' +
                        '<td id="IntViolation_input_tariff_' + d.ret + '">' + d.full_data.input_tariff + '</td>' +
                        '<td id="IntViolation_good_count_' + d.ret + '">' + d.full_data.good_count + '</td>' +
                        '<td id="IntViolation_price_' + d.ret + '">' + d.full_data.price + '</td>' +
                        '<td id="IntViolation_truck_no_' + d.ret + '">' + d.full_data.truck_no + '</td>' +
                        '<td id="IntViolation_driver_name_' + d.ret + '">' + d.full_data.driver_name + '</td>' +
                        '<td id="IntViolation_chasis_no_' + d.ret + '">' + d.full_data.chasis_no + '</td>' +
                        '<td id="IntViolation_trailer_no_' + d.ret + '">' + d.full_data.trailer_no + '</td>' +
                        '<td id="IntViolation_tranian_truck_' + d.ret + '">' + d.full_data.tranian_truck + '</td>' +
                        '<td id="IntViolation_truck_address_' + d.ret + '">' + d.full_data.truck_address + '</td>' +
                        '<td id="IntViolation_truck_type_' + d.ret + '">' + d.full_data.truck_type + '</td>' +
                        '<td id="IntViolation_vi_type_1_' + d.ret + '">' + d.full_data.vi_type_1 + '</td>' +
                        '<td id="IntViolation_vi_type_2_' + d.ret + '">' + d.full_data.vi_type_2 + '</td>' +
                        '<td id="IntViolation_vi_type_3_' + d.ret + '">' + d.full_data.vi_type_3 + '</td>' +
                        '<td id="IntViolation_vi_type_4_' + d.ret + '">' + d.full_data.vi_type_4 + '</td>' +
                        '<td id="IntViolation_vi_type_5_' + d.ret + '">' + d.full_data.vi_type_5 + '</td>' +
                        '<td id="IntViolation_vi_type_6_' + d.ret + '">' + d.full_data.vi_type_6 + '</td>' +
                        '<td id="IntViolation_vi_type_7_' + d.ret + '">' + d.full_data.vi_type_7 + '</td>' +
                        '<td id="IntViolation_detector_' + d.ret + '">' + d.full_data.detector + '</td>' +
                        '<td id="IntViolation_detector_other_' + d.ret + '">' + d.full_data.detector_other + '</td>' +
                        '<td id="IntViolation_fine_receipt_no_' + d.ret + '">' + d.full_data.fine_receipt_no + '</td>' +
                        '<td id="IntViolation_fine_price_' + d.ret + '">' + d.full_data.fine_price + '</td>' +
                        '<td id="IntViolation_final_result_' + d.ret + '">' + d.full_data.final_result + '</td>' +
                        '<td id="IntViolation_details_' + d.ret + '">' + d.full_data.details + '</td>' +
                        '<td id="IntViolation_m_rec_date_' + d.ret + '">' + d.full_data.m_rec_date + '</td>' +
                        '<td id="IntViolation_s_rec_date_' + d.ret + '">' + d.full_data.s_rec_date + '</td>' +
                        '<td id="IntViolation_locked_' + d.ret + '">' + d.full_data.locked + '</td>' +

                        '<td>' +
                        '   <span class="btn btn-warning" onclick="edit_IntViolation(\'' + d.ret + '\')">Edit</span>' +
                        '   <span class="btn btn-danger" onclick="remove_IntViolation(\'' + d.ret + '\')">Remove</span>' +
                        '</td>' +
                        '</tr>';
                $('#table_IntViolation tbody').append(td);
                break;
            case 'edit':
                $('#IntViolation_kootaj_' + d.ret).html(d.full_data.kootaj);
                $('#IntViolation_cert_no_' + d.ret).html(d.full_data.cert_no);
                $('#IntViolation_s_date_' + d.ret).html(d.full_data.s_date);
                $('#IntViolation_m_date_' + d.ret).html(d.full_data.m_date);
                $('#IntViolation_int_cert_no_' + d.ret).html(d.full_data.int_cert_no);
                $('#IntViolation_radif_marzi_' + d.ret).html(d.full_data.radif_marzi);
                $('#IntViolation_s_entry_date_' + d.ret).html(d.full_data.s_entry_date);
                $('#IntViolation_m_entry_date_' + d.ret).html(d.full_data.m_entry_date);
                $('#IntViolation_karne_tir_no_' + d.ret).html(d.full_data.karne_tir_no);
                $('#IntViolation_ezhar_1_' + d.ret).html(d.full_data.ezhar_1);
                $('#IntViolation_ezhar_2_' + d.ret).html(d.full_data.ezhar_2);
                $('#IntViolation_ezhar_3_' + d.ret).html(d.full_data.ezhar_3);
                $('#IntViolation_ezhar_4_' + d.ret).html(d.full_data.ezhar_4);
                $('#IntViolation_person_type_' + d.ret).html(d.full_data.person_type);
                $('#IntViolation_full_name_' + d.ret).html(d.full_data.full_name);
                $('#IntViolation_card_no_' + d.ret).html(d.full_data.card_no);
                $('#IntViolation_code_no_' + d.ret).html(d.full_data.code_no);
                $('#IntViolation_dest_address_' + d.ret).html(d.full_data.dest_address);
                $('#IntViolation_good_bame_' + d.ret).html(d.full_data.good_bame);
                $('#IntViolation_net_weight_' + d.ret).html(d.full_data.net_weight);
                $('#IntViolation_weight_' + d.ret).html(d.full_data.weight);
                $('#IntViolation_input_tariff_' + d.ret).html(d.full_data.input_tariff);
                $('#IntViolation_good_count_' + d.ret).html(d.full_data.good_count);
                $('#IntViolation_price_' + d.ret).html(d.full_data.price);
                $('#IntViolation_truck_no_' + d.ret).html(d.full_data.truck_no);
                $('#IntViolation_driver_name_' + d.ret).html(d.full_data.driver_name);
                $('#IntViolation_chasis_no_' + d.ret).html(d.full_data.chasis_no);
                $('#IntViolation_trailer_no_' + d.ret).html(d.full_data.trailer_no);
                $('#IntViolation_tranian_truck_' + d.ret).html(d.full_data.tranian_truck);
                $('#IntViolation_truck_address_' + d.ret).html(d.full_data.truck_address);
                $('#IntViolation_truck_type_' + d.ret).html(d.full_data.truck_type);
                $('#IntViolation_vi_type_1_' + d.ret).html(d.full_data.vi_type_1);
                $('#IntViolation_vi_type_2_' + d.ret).html(d.full_data.vi_type_2);
                $('#IntViolation_vi_type_3_' + d.ret).html(d.full_data.vi_type_3);
                $('#IntViolation_vi_type_4_' + d.ret).html(d.full_data.vi_type_4);
                $('#IntViolation_vi_type_5_' + d.ret).html(d.full_data.vi_type_5);
                $('#IntViolation_vi_type_6_' + d.ret).html(d.full_data.vi_type_6);
                $('#IntViolation_vi_type_7_' + d.ret).html(d.full_data.vi_type_7);
                $('#IntViolation_detector_' + d.ret).html(d.full_data.detector);
                $('#IntViolation_detector_other_' + d.ret).html(d.full_data.detector_other);
                $('#IntViolation_fine_receipt_no_' + d.ret).html(d.full_data.fine_receipt_no);
                $('#IntViolation_fine_price_' + d.ret).html(d.full_data.fine_price);
                $('#IntViolation_final_result_' + d.ret).html(d.full_data.final_result);
                $('#IntViolation_details_' + d.ret).html(d.full_data.details);
                $('#IntViolation_m_rec_date_' + d.ret).html(d.full_data.m_rec_date);
                $('#IntViolation_s_rec_date_' + d.ret).html(d.full_data.s_rec_date);
                $('#IntViolation_locked_' + d.ret).html(d.full_data.locked);

                break;
            case 'delete':
                $('tr#IntViolation_' + d.ret).remove();
                var i = 1;
                $('#table_IntViolation tbody tr').each(function(){
                    $('td:nth-child(1)',this).html(i++);
                });
                break;
			case 'info':
                $('#form_action').val('edit');
                $('#IntViolation_id').val(d.ret);
				
                $('#IntViolation_kootaj').val(d.full_data.kootaj);
                $('#IntViolation_cert_no').val(d.full_data.cert_no);
                $('#IntViolation_s_date').val(d.full_data.s_date);
                $('#IntViolation_m_date').val(d.full_data.m_date);
                $('#IntViolation_int_cert_no').val(d.full_data.int_cert_no);
                $('#IntViolation_radif_marzi').val(d.full_data.radif_marzi);
                $('#IntViolation_s_entry_date').val(d.full_data.s_entry_date);
                $('#IntViolation_m_entry_date').val(d.full_data.m_entry_date);
                $('#IntViolation_karne_tir_no').val(d.full_data.karne_tir_no);
                $('#IntViolation_ezhar_1').val(d.full_data.ezhar_1);
                $('#IntViolation_ezhar_2').val(d.full_data.ezhar_2);
                $('#IntViolation_ezhar_3').val(d.full_data.ezhar_3);
                $('#IntViolation_ezhar_4').val(d.full_data.ezhar_4);
                $('#IntViolation_person_type').val(d.full_data.person_type);
                $('#IntViolation_full_name').val(d.full_data.full_name);
                $('#IntViolation_card_no').val(d.full_data.card_no);
                $('#IntViolation_code_no').val(d.full_data.code_no);
                $('#IntViolation_dest_address').val(d.full_data.dest_address);
                $('#IntViolation_good_bame').val(d.full_data.good_bame);
                $('#IntViolation_net_weight').val(d.full_data.net_weight);
                $('#IntViolation_weight').val(d.full_data.weight);
                $('#IntViolation_input_tariff').val(d.full_data.input_tariff);
                $('#IntViolation_good_count').val(d.full_data.good_count);
                $('#IntViolation_price').val(d.full_data.price);
                $('#IntViolation_truck_no').val(d.full_data.truck_no);
                $('#IntViolation_driver_name').val(d.full_data.driver_name);
                $('#IntViolation_chasis_no').val(d.full_data.chasis_no);
                $('#IntViolation_trailer_no').val(d.full_data.trailer_no);
                $('#IntViolation_tranian_truck').val(d.full_data.tranian_truck);
                $('#IntViolation_truck_address').val(d.full_data.truck_address);
                $('#IntViolation_truck_type').val(d.full_data.truck_type);
                $('#IntViolation_vi_type_1').val(d.full_data.vi_type_1);
                $('#IntViolation_vi_type_2').val(d.full_data.vi_type_2);
                $('#IntViolation_vi_type_3').val(d.full_data.vi_type_3);
                $('#IntViolation_vi_type_4').val(d.full_data.vi_type_4);
                $('#IntViolation_vi_type_5').val(d.full_data.vi_type_5);
                $('#IntViolation_vi_type_6').val(d.full_data.vi_type_6);
                $('#IntViolation_vi_type_7').val(d.full_data.vi_type_7);
                $('#IntViolation_detector').val(d.full_data.detector);
                $('#IntViolation_detector_other').val(d.full_data.detector_other);
                $('#IntViolation_fine_receipt_no').val(d.full_data.fine_receipt_no);
                $('#IntViolation_fine_price').val(d.full_data.fine_price);
                $('#IntViolation_final_result').val(d.full_data.final_result);
                $('#IntViolation_details').val(d.full_data.details);
                $('#IntViolation_m_rec_date').val(d.full_data.m_rec_date);
                $('#IntViolation_s_rec_date').val(d.full_data.s_rec_date);
                $('#IntViolation_locked').val(d.full_data.locked);

                break;
        }
    }).fail(function(){
        alert('Error')
    });
    reset_IntViolation();
}

File: views/int_violation.py
import tornado.httpserver
import tornado.ioloop
import tornado.options
import tornado.web
from controllers.int_violation import Intviolation as Controller
from tools.redis import Redis


class Intviolation(tornado.web.RequestHandler):

    def get(self, *args, **kwargs):
        acc_type = Redis.get(key='acc_type')
        if acc_type == 'admin':
            permissions = Redis.get(key='permissions', type='list')
            self.render('IntViolation.html', permissions=permissions)
        else:
            # self.write('Data')
            # self.write(acc_type)
            self.redirect('/login')

    def post(self, *args, **kwargs):

        id = self.get_argument('id','')
        kootaj = self.get_argument('kootaj', '')
        cert_no = self.get_argument('cert_no', '')
        s_date = self.get_argument('s_date', '')
        m_date = self.get_argument('m_date', '')
        int_cert_no = self.get_argument('int_cert_no', '')
        radif_marzi = self.get_argument('radif_marzi', '')
        s_entry_date = self.get_argument('s_entry_date', '')
        m_entry_date = self.get_argument('m_entry_date', '')
        karne_tir_no = self.get_argument('karne_tir_no', '')
        ezhar_1 = self.get_argument('ezhar_1', False)
        ezhar_2 = self.get_argument('ezhar_2', False)
        ezhar_3 = self.get_argument('ezhar_3', False)
        ezhar_4 = self.get_argument('ezhar_4', False)
        person_type = self.get_argument('person_type', False)
        full_name = self.get_argument('full_name', '')
        card_no = self.get_argument('card_no', '')
        code_no = self.get_argument('code_no', '')
        dest_address = self.get_argument('dest_address', '')
        good_name = self.get_argument('good_name', '')
        net_weight = self.get_argument('net_weight', '')
        weight = self.get_argument('weight', '')
        input_tariff = self.get_argument('input_tariff', '')
        good_count = self.get_argument('good_count', 0)
        price = self.get_argument('price', 0)
        truck_no = self.get_argument('truck_no', '')
        driver_name = self.get_argument('driver_name', '')
        chasis_no = self.get_argument('chasis_no', '')
        trailer_no = self.get_argument('trailer_no', '')
        iranian_truck = self.get_argument('iranian_truck', False)
        truck_address = self.get_argument('truck_address', '')
        truck_type = self.get_argument('truck_type', 0)
        vi_type_1 = self.get_argument('vi_type_1', False)
        vi_type_2 = self.get_argument('vi_type_2', False)
        vi_type_3 = self.get_argument('vi_type_3', False)
        vi_type_4 = self.get_argument('vi_type_4', False)
        vi_type_5 = self.get_argument('vi_type_5', False)
        vi_type_6 = self.get_argument('vi_type_6', False)
        vi_type_7 = self.get_argument('vi_type_7', False)
        detector = self.get_argument('detector', 0)
        detector_other = self.get_argument('detector_other', '')
        fine_receipt_no = self.get_argument('fine_receipt_no', '')
        fine_price = self.get_argument('fine_price', 0)
        final_result = self.get_argument('final_result', 0)
        details = self.get_argument('details', '')
        m_rec_date = self.get_argument('m_rec_date', '')
        s_rec_date = self.get_argument('s_rec_date', '')
        locked = self.get_argument('locked', False)

        # self.write('kootaj:' + str(kootaj)+'<br>')
        # self.write('cert_no:' + str(cert_no)+'<br>')
        # self.write('s_date:' + str(s_date)+'<br>')
        # self.write('m_date:' + str(m_date)+'<br>')
        # self.write('int_cert_no:' + str(int_cert_no)+'<br>')
        # self.write('radif_marzi:' + str(radif_marzi)+'<br>')
        # self.write('s_entry_date:' + str(s_entry_date)+'<br>')
        # self.write('m_entry_date:' + str(m_entry_date)+'<br>')
        # self.write('karne_tir_no:' + str(karne_tir_no)+'<br>')
        # self.write('ezhar_1:' + str(ezhar_1)+'<br>')
        # self.write('ezhar_2:' + str(ezhar_2)+'<br>')
        # self.write('ezhar_3:' + str(ezhar_3)+'<br>')
        # self.write('ezhar_4:' + str(ezhar_4)+'<br>')
        # self.write('person_type:' + str(person_type)+'<br>')
        # self.write('full_name:' + str(full_name)+'<br>')
        # self.write('card_no:' + str(card_no)+'<br>')
        # self.write('code_no:' + str(code_no)+'<br>')
        # self.write('dest_address:' + str(dest_address)+'<br>')
        # self.write('good_name:' + str(good_name)+'<br>')
        # self.write('net_weight:' + str(net_weight)+'<br>')
        # self.write('weight:' + str(weight)+'<br>')
        # self.write('input_tariff:' + str(input_tariff)+'<br>')
        # self.write('good_count:' + str(good_count)+'<br>')
        # self.write('price:' + str(price)+'<br>')
        # self.write('truck_no:' + str(truck_no)+'<br>')
        # self.write('driver_name:' + str(driver_name)+'<br>')
        # self.write('chasis_no:' + str(chasis_no)+'<br>')
        # self.write('trailer_no:' + str(trailer_no)+'<br>')
        # self.write('iranian_truck:' + str(iranian_truck)+'<br>')
        # self.write('truck_address:' + str(truck_address)+'<br>')
        # self.write('truck_type:' + str(truck_type)+'<br>')
        # self.write('vi_type_1:' + str(vi_type_1)+'<br>')
        # self.write('vi_type_2:' + str(vi_type_2)+'<br>')
        # self.write('vi_type_3:' + str(vi_type_3)+'<br>')
        # self.write('vi_type_4:' + str(vi_type_4)+'<br>')
        # self.write('vi_type_5:' + str(vi_type_5)+'<br>')
        # self.write('vi_type_6:' + str(vi_type_6)+'<br>')
        # self.write('vi_type_7:' + str(vi_type_7)+'<br>')
        # self.write('detector:' + str(detector)+'<br>')
        # self.write('detector_other:' + str(detector_other)+'<br>')
        # self.write('fine_receipt_no:' + str(fine_receipt_no)+'<br>')
        # self.write('fine_price:' + str(fine_price)+'<br>')
        # self.write('final_result:' + str(final_result)+'<br>')
        # self.write('details:' + str(details)+'<br>')
        # self.write('m_rec_date:' + str(m_rec_date)+'<br>')
        # self.write('s_rec_date:' + str(s_rec_date)+'<br>')
        # self.write('locked:' + str(locked)+'<br>')

        Controller.add(id=id, kootaj=kootaj, cert_no=cert_no, s_date=s_date, m_date=m_date, int_cert_no=int_cert_no,
                       radif_marzi=radif_marzi, s_entry_date=s_entry_date, m_entry_date=m_entry_date,
                       karne_tir_no=karne_tir_no, ezhar_1=ezhar_1, ezhar_2=ezhar_2, ezhar_3=ezhar_3, ezhar_4=ezhar_4,
                       person_type=person_type, full_name=full_name, card_no=card_no, code_no=code_no,
                       dest_address=dest_address, good_name=good_name, net_weight=net_weight, weight=weight,
                       input_tariff=input_tariff, good_count=good_count, price=price, truck_no=truck_no,
                       driver_name=driver_name, chasis_no=chasis_no, trailer_no=trailer_no, iranian_truck=iranian_truck,
                       truck_address=truck_address, truck_type=truck_type, vi_type_1=vi_type_1, vi_type_2=vi_type_2,
                       vi_type_3=vi_type_3, vi_type_4=vi_type_4, vi_type_5=vi_type_5, vi_type_6=vi_type_6,
                       vi_type_7=vi_type_7, detector=detector, detector_other=detector_other,
                       fine_receipt_no=fine_receipt_no, fine_price=fine_price, final_result=final_result,
                       details=details, m_rec_date=m_rec_date, s_rec_date=s_rec_date, locked=locked)
        self.write(str(vi_type_4))
        self.write(str(vi_type_3))
        # self.redirect('/int_violation')
